File: content/code/rlnc/equivalent.java
public interface Bonjour {
    public void dire_bonjour();
}

public class Personne implements Bonjour {
    public String name;

    public void dire_bonjour() {
        System.out.println("Bonjour, je m'appelle " + this.name);
    }

    public Personne(String name) {
        this.name = name;
    }
}

public class equivalent {
    public static void main(String[] args) {
        Personne p = new Personne("John Smith");
        p.dire_bonjour();
    }
}

File: content/code/rlnc/equivalent.rs
struct Personne {
    name: String,
}

trait Bonjour {
    fn dire_bonjour(&self);
}

impl Bonjour for Personne {
    fn dire_bonjour(&self) {
        println!("Bonjour, je m'appelle {}", self.name);
    }
}

fn main() {
    let p: Personne = Personne {
        name: String::from("John Smith"),
    };

    p.dire_bonjour();
}
